Sources/BrowserServicesKit/Suggestions/SuggestionLoading.swift

```diff
@@ -38,9 +40,11 @@ public class SuggestionLoader: SuggestionLoading {
     }
 
     public weak var dataSource: SuggestionLoadingDataSource?
+    private var urlFactory: ((String) -> URL?)?
 
-    public init(dataSource: SuggestionLoadingDataSource? = nil) {
+    public init(dataSource: SuggestionLoadingDataSource? = nil, urlFactory: ((String) -> URL?)? = nil) {
```

Comment:
Sorry, just few details that got my attention. What is the reason to have the urlFactory as optional? I can imagine it can be overlooked with the nil as default argument and the library will basically not work correctly.

Reply:
This is done to keep the API backwards compatible

Reply:
It's pretty early in the life of this library and the usage of it, let's not worry about that so much.  Breaking API changes just mean bumping the major version so if we rapidly move from 1.0.1 to 2.0.0 then that's fine, that's the semver.org way and nothing to worry about. :)